### DISCUSSION TITLE
About the uncontrollable problems occurred during Krylov subspace reduction

### DISCUSSION THREAD

**Discussion by Yang1816910001** [ImperialCollegeLondon/sharpy#274] [Q&A]
After getting state-space of the coupled linear reduced model from SHARPy and adjust it into a model for controller design, I found that some Krylov subspace states are uncontrollable through checking the ctrb matrix of A and B,  and similar problem was mentioned in PHD thesis of Dr.Atrola, while some paper using LQR or other way seldom talks the solution about it ,by the way, I am also wondering how to assemble the linear UVLM model with nonlinear model.
so I have questions below:
1. Can problems like partial states uncontrollable be solved through parameter tuning? And how to solve this problem in most of your team's research.
2. How to assemble the reduced linear UVLM with nonlinear structural model, through functions in SHARPy or other ways?
I am still reading thesis written by your teams, so problems here are not that professional and ripe, but I'm really curious about how to solve that, it would be so kind for any help you provided

**Comment by rafapalacios** [comment]
1. Krylov subspace ROM is not stability preserving, so there is no straightforward solution to this. Consider modifying the discretization in your model and the size of the ROM. You can also try the balancing methods in SHARPy.
2.This is not a feature in SHARPy.

**Reply by Yang1816910001** [reply]
Thanks a lot, and is there any cod examples for assembling the UVLM model with the nonlinear one?
